### PR TITLE
[REVIEW] Unpin `dask` and `distributed` in CI

### PR DESCRIPTION
**Pull request by galipremsagar**
This PR unpins `dask` and `distributed` in CI.

### CODE REVIEW

**Review by charlesbluca — COMMENTED** [review]
LGTM; could you point me to an issue / CI failure motivating this change?

File: conda/environments/cudf_dev_cuda11.0.yml [hunk 1]
```diff
@@ -39,8 +39,8 @@ dependencies:
   - mypy=0.782
   - typing_extensions
   - pre_commit
-  - dask=2021.09.1
-  - distributed=2021.09.1
+  - dask>=2021.09.1,<2021.10.0
+  - distributed>=2021.09.1,<2021.10.0
```
Comment (galipremsagar) [inline conda/environments/cudf_dev_cuda11.0.yml:43]:
```suggestion
  - dask>=2021.09.1
  - distributed>=2021.09.1
```

File: conda/environments/cudf_dev_cuda11.2.yml [hunk 2]
```diff
@@ -39,8 +39,8 @@ dependencies:
   - mypy=0.782
   - typing_extensions
   - pre_commit
-  - dask=2021.09.1
-  - distributed=2021.09.1
+  - dask>=2021.09.1,<2021.10.0
+  - distributed>=2021.09.1,<2021.10.0
```
Comment (galipremsagar) [inline conda/environments/cudf_dev_cuda11.2.yml:43]:
```suggestion
  - dask>=2021.09.1
  - distributed>=2021.09.1
```

File: conda/recipes/custreamz/meta.yaml [hunk 3]
```diff
@@ -31,8 +31,8 @@ requirements:
     - python
     - streamz 
     - cudf {{ version }}
-    - dask=2021.09.1
-    - distributed=2021.09.1
+    - dask>=2021.09.1,<2021.10.0
+    - distributed>=2021.09.1,<2021.10.0
```
Comment (galipremsagar) [inline conda/recipes/custreamz/meta.yaml:35]:
```suggestion
    - dask>=2021.09.1
    - distributed>=2021.09.1
```

File: conda/recipes/dask-cudf/meta.yaml [hunk 4]
```diff
@@ -26,13 +26,13 @@ requirements:
   host:
     - python
     - cudf {{ version }}
-    - dask=2021.09.1
-    - distributed=2021.09.1
+    - dask>=2021.09.1,<2021.10.0
+    - distributed>=2021.09.1,<2021.10.0
```
Comment (galipremsagar) [inline conda/recipes/dask-cudf/meta.yaml:30]:
```suggestion
    - dask>=2021.09.1
    - distributed>=2021.09.1
```

File: conda/recipes/dask-cudf/meta.yaml [hunk 5]
```diff
@@ -26,13 +26,13 @@ requirements:
...
   run:
     - python
     - cudf {{ version }}
-    - dask=2021.09.1
-    - distributed=2021.09.1
+    - dask>=2021.09.1,<2021.10.0
+    - distributed>=2021.09.1,<2021.10.0
```
Comment (galipremsagar) [inline conda/recipes/dask-cudf/meta.yaml:35]:
```suggestion
    - dask>=2021.09.1
    - distributed>=2021.09.1
```

File: python/custreamz/dev_requirements.txt [hunk 6]
```diff
@@ -3,8 +3,8 @@
 flake8==3.8.3
 black==19.10b0
 isort==5.6.4
-dask==2021.09.1
-distributed==2021.09.1
+dask>=2021.09.1,<2021.10.0
+distributed>=2021.09.1,<2021.10.0
```
Comment (galipremsagar) [inline python/custreamz/dev_requirements.txt:7]:
```suggestion
dask>=2021.09.1
distributed>=2021.09.1
```

File: python/dask_cudf/dev_requirements.txt [hunk 7]
```diff
@@ -1,7 +1,7 @@
 # Copyright (c) 2021, NVIDIA CORPORATION.
 
-dask==2021.09.1
-distributed==2021.09.1
+dask>=2021.09.1,<2021.10.0
+distributed>=2021.09.1,<2021.10.0
```
Comment (galipremsagar) [inline python/dask_cudf/dev_requirements.txt:4]:
```suggestion
dask>=2021.09.1
distributed>=2021.09.1
```

File: python/dask_cudf/setup.py [hunk 8]
```diff
@@ -10,8 +10,8 @@
 
 install_requires = [
     "cudf",
-    "dask==2021.09.1",
-    "distributed==2021.09.1",
+    "dask>=2021.09.1,<2021.10.0",
+    "distributed>=2021.09.1,<2021.10.0",
```
Comment (galipremsagar) [inline python/dask_cudf/setup.py:14]:
```suggestion
    "dask>=2021.09.1",
    "distributed>=2021.09.1",
```

File: python/dask_cudf/setup.py [hunk 9]
```diff
@@ -23,8 +23,8 @@
         "pandas>=1.0,<1.4.0dev0",
         "pytest",
         "numba>=0.53.1",
-        "dask==2021.09.1",
-        "distributed==2021.09.1",
+        "dask>=2021.09.1,<2021.10.0",
+        "distributed>=2021.09.1,<2021.10.0",
```
Comment (galipremsagar) [inline python/dask_cudf/setup.py:27]:
```suggestion
        "dask>=2021.09.1",
        "distributed>=2021.09.1",
```

**Comment by galipremsagar** [comment]
> LGTM; could you point me to an issue / CI failure motivating this change?

No issue or CI failure that triggered this PR. 

We usually pin and restrict our users to certain versions of `dask` that are released only until the time of release, i.e., in this case for `21.10` we pinned the dask version to `2021.9.1`, but we will need to unpin in `21.12` since it is now in development phase and then pin the dask version again during the release of `21.12`.

**Review by charlesbluca — APPROVED** [review]
Thanks for the the clarification @galipremsagar 🙂

**Review by ajschmidt8 — APPROVED** [review]

**Comment by shwina** [comment]
@gpucibot merge

**Review by jakirkham — APPROVED** [review]

**Pull request merged**